certified-connectors/Cognite Data Fusion/apiProperties.json

```diff
@@ -7,29 +7,166 @@
       "token": {
         "type": "oauthSetting",
         "oAuthSettings": {
-          "identityProvider": "oauth2",
+          "identityProvider": "oauth2generic",
           "clientId": "PLACEHOLDER_CLIENTID",
-          "scopes": ["https://bluefield.cognitedata.com/.default"],
-          "redirectMode": "Global",
-          "redirectUrl": "https://global.consent.azure-apim.net/redirect",
+          "scopes": ["{Scope}"],
+          "redirectMode": "GlobalPerConnector",
           "properties": {
-            "IsFirstParty": "False"
+            "IsFirstParty": "False",
+            "IsOnbehalfofLoginSupported": true
           },
           "customParameters": {
-            "authorizationUrl": {
-              "value": "https://login.microsoftonline.com/PLACEHOLDER_TENANTID/oauth2/v2.0/authorize"
+            "authorizationUrlTemplate": {
+              "value": "{AuthorizationUrl}"
             },
-            "tokenUrl": {
-              "value": "https://login.microsoftonline.com/PLACEHOLDER_TENANTID/oauth2/v2.0/token"
+            "authorizationUrlQueryStringTemplate":{
+               "value":"client_id={ClientId}&response_type=code&scope={Scope}&state={State}"
             },
-            "refreshUrl": {
-              "value": "https://login.microsoftonline.com/PLACEHOLDER_TENANTID/oauth2/v2.0/token"
+            "tokenUrlTemplate": {
+              "value": "{TokenUrl}"
+            },
+            "tokenBodyTemplate":{
+               "value":"code={code}&grant_type=authorization_code&client_id={ClientId}&client_secret={ClientSecret}&scope={Scope}"
+            },
+            "refreshUrlTemplate": {
+              "value": "{RefreshUrl}"
+            },
+            "refreshBodyTemplate":{
+               "value":"code={code}&grant_type=authorization_code&client_id={ClientId}&client_secret={ClientSecret}&scope={Scope}"
             }
           }
+        },
+        "uiDefinition": {
+					"displayName": "Login",
+					"description": "Sign in with your Cognite Data Fusion credentials",
+					"tooltip": "Sign in with your Cognite Data Fusion credentials",
+					"constraints": {
+						"required": "true"
+					}
+				}
+      },
+      "token:CDFCluster": {
+        "type": "string",
+        "uiDefinition": {
+          "constraints": {
+            "required": "true"
+          },
+          "displayName": "Enter your Cognite Data Fusion Cluster.",
+          "description": "Enter your Cognite Data Fusion Cluster.",
+          "tooltip": "api, greenfield, bluefield, az-eastus-1, etc..."
+        }
+      },
+      "token:TenantId": {
+        "metadata": {
+          "sourceType": "AzureActiveDirectoryTenant"
+        },
+        "type": "string",
+        "uiDefinition": {
+          "constraints": {
+            "hidden": "true",
+            "required": "false"
+          },
+          "description": "The tenant ID of for the Azure Active Directory application",
```

Comment:
The apiproperties.json validation is still the error. Could you please remove description and display properties?

##[error]@{level=Critical; message=Invalid value for the description property within the UI Definition object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.description; value=The tenant ID of for the Azure Active Directory application}
##[error]@{level=Critical; message=Invalid value for the display name property within the UI Defintion object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.displayName; value=Tenant}